pkg/kubeapiserver/admission/util/pod.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// IsUpdatingInitializedPod returns if operation is trying to update an uninitialized pod.
+func IsUpdatingInitializedPod(a admission.Attributes) bool {
+	if a.GetOperation() != admission.Update {
+		return false
+	}
+	oldObj := a.GetOldObject()
+	oldPod, ok := oldObj.(*api.Pod)
+	if !ok {
+		return false
+	}
+	if oldPod.Initializers == nil {
+		return true
+	}
+	return false
+}
+
+// IsUpdatingUninitializedPod returns if operation is trying to update an uninitialized pod.
+func IsUpdatingUninitializedPod(a admission.Attributes) bool {
```

Comment:
This should be completely generic instead of pod specific.  Use meta.GetObjectMeta() and check Initializers generically.